src/runtime/set.cpp

```diff
@@ -600,20 +642,121 @@ Box* setGt(BoxedSet* self, BoxedSet* rhs) {
     return setIssuperset(self, rhs);
 }
 
+Box* setContains(BoxedSet* self, Box* key) {
+    RELEASE_ASSERT(PyAnySet_Check(self), "");
+
+    if (PySet_Check(key)) {
+        try {
+            BoxAndHash k_hash(key);
+            return boxBool(self->s.find(k_hash) != self->s.end());
+        } catch (ExcInfo e) {
+            if (e.matches(TypeError))
+                PyErr_Clear();
+            else {
+                throw e;
+            }
+            BoxedSet* tmpKey = makeNewSet(frozenset_cls, key);
+            return boxBool(self->s.find(tmpKey) != self->s.end());
+        }
+    }
+
+    return boxBool(self->s.find(key) != self->s.end());
+}
+
+Box* setRemove(BoxedSet* self, Box* key) {
+    RELEASE_ASSERT(isSubclass(self->cls, set_cls), "");
+
+    if (PySet_Check(key)) {
+        try {
+            BoxAndHash k_hash(key);
+            if (self->s.find(k_hash) != self->s.end()) {
```

Comment:
FYI `erase()` returns a bool, which will be true if the element existed or not.  It's probably not a big deal and I bet the compiler can figure a lot of this out, but I think it's slightly better form to do `bool existed = self->s.erase(k_hash); if (existed) return None;`